sdk/spring-sdk/src/test/scala/kalix/springsdk/impl/SpringSDKTestRunner.scala

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright 2021 Lightbend Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package kalix.springsdk.impl
+
+import kalix.javasdk.Kalix
+import kalix.springsdk.action.EchoAction
+import kalix.springsdk.action.ReflectiveActionProvider
+
+// Temporary class just for testing
+object SpringSDKTestRunner extends App {
+  val kalix = new Kalix()
+  kalix.register(ReflectiveActionProvider.of(classOf[EchoAction], _ => new EchoAction))
+  kalix.start().toCompletableFuture.get()
```

Comment:
Refactor this test runner to use the register method. This is important because the previous version was not detecting an issue with the `ResolvedActionFactory`. 

The issue is now solved because we won't use a `ResolvedActionFactory` when the provider has a alternative `MessageCodec`.